parquet-variant-compute/src/variant_parser.rs

```diff
@@ -0,0 +1,669 @@
+                size
+            ))),
+        }
+    }
+
+    /// Calculate the size needed to store an integer
+    pub fn calculate_int_size(value: usize) -> usize {
+        if value <= u8::MAX as usize {
+            1
+        } else if value <= u16::MAX as usize {
+            2
+        } else if value <= 0xFFFFFF {
+            3
+        } else {
+            4
+        }
+    }
+
+    /// Build object header byte
+    pub fn build_object_header(
+        is_large: bool,
+        field_id_size: usize,
+        field_offset_size: usize,
+    ) -> u8 {
+        let large_bit = if is_large { 1 } else { 0 };
+        (large_bit << 6)
+            | (((field_id_size - 1) as u8) << 4)
+            | (((field_offset_size - 1) as u8) << 2)
+            | 2
+    }
+
+    /// Write integer bytes to buffer
+    pub fn write_int_bytes(buffer: &mut Vec<u8>, value: usize, size: usize) {
+        match size {
+            1 => buffer.push(value as u8),
+            2 => buffer.extend_from_slice(&(value as u16).to_le_bytes()),
+            3 => {
+                let bytes = (value as u32).to_le_bytes();
+                buffer.extend_from_slice(&bytes[..3]);
+            }
+            4 => buffer.extend_from_slice(&(value as u32).to_le_bytes()),
+            _ => panic!("Invalid size: {}", size),
+        }
+    }
+
+
+
+    /// Check if value bytes represent a primitive
+    pub fn is_primitive(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::Primitive
+    }
+
+    /// Check if value bytes represent a short string
+    pub fn is_short_string(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::ShortString
+    }
+
+    /// Check if value bytes represent an object
+    pub fn is_object(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::Object
+    }
+
+    /// Check if value bytes represent an array
+    pub fn is_array(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::Array
+    }
+
+    /// Get the data length for a primitive type
+    /// Returns Some(len) for fixed-length types, None for variable-length types
+    pub fn get_primitive_data_length(primitive_type: &PrimitiveType) -> Option<usize> {
+        match primitive_type {
+            PrimitiveType::Null | PrimitiveType::True | PrimitiveType::False => Some(0),
+            PrimitiveType::Int8 => Some(1),
+            PrimitiveType::Int16 => Some(2),
+            PrimitiveType::Int32
+            | PrimitiveType::Float
+            | PrimitiveType::Decimal4
+            | PrimitiveType::Date => Some(4),
```

Comment:
might be worth a `use`?
```suggestion
        use PrimitiveType::*;
        match primitive_type {
            Null | True | False => Some(0),
            Int8 => Some(1),
            Int16 => Some(2),
            Int32 | Float | Decimal4 | Date => Some(4),
```